java/arcs/core/entity/EntityBase.kt

```diff
@@ -242,15 +242,21 @@ open class EntityBase(
         schema.fields.collections.keys.forEach { collections[it] = emptySet() }
     }
 
-    override fun serialize(): RawEntity {
+    override fun serialize(restrictedSchema: Schema?): RawEntity {
+        val singletonFields = if (restrictedSchema != null) {
+            restrictedSchema.fields.singletons
+        } else schema.fields.singletons
+        val collectionFields = if (restrictedSchema != null) {
+            restrictedSchema.fields.collections
+        } else schema.fields.collections
```

Comment:
Could this be simplified to:
`val serializationSchema = restrctedSchema ?: schema`

and then using below?
`singletons = serializationSchema.singletons.mapValues` 

etc?

Reply:
done, thank you!